File: Data Structures & Algorithms/majority-element/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        myMap = {}
        for num in nums:
            if num in myMap:
                myMap[num] = 1 + myMap[num]
            else:
                myMap[num] = 1

        (n, t) = (-1, -1)
        for num in myMap:
            if myMap[num] > t:
                (n, t) = (num, myMap[num]) 
        return n

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-0.java
class Solution {
    public int findKthLargest(int[] nums, int k) {

        PriorityQueue<Integer> minHeap = new PriorityQueue<>((a , b) -> b - a);

        for ( int num : nums) {
            minHeap.offer(num);
        }

        for (int i = 0; i < k - 1; i++) {
            minHeap.poll();
        }
        return minHeap.peek();
    }
}


// class Solution {
//     public int findKthLargest(int[] nums, int k) {

//         Arrays.sort(nums);
//         return nums[nums.length - k];
        
//     }
// }
